tsdb/db.go

```diff
@@ -1880,11 +1898,23 @@ func (db *DB) Querier(mint, maxt int64) (storage.Querier, error) {
 			blocks = append(blocks, b)
 		}
 	}
-	var inOrderHeadQuerier storage.Querier
+
+	blockQueriers := make([]storage.Querier, 0, len(blocks)+2) // +2 to allow for possible in-order and OOO head queriers
+
+	defer func() {
```

Comment:
I like shifting the defer to the beginning 👍  I'm just slightly worried that the err gets shadowed, if you look below we're declaring an err on each of the if code blocks but this err actually refers to the last `for` where we are iterating the list of TSDB blocks... so either I'd colocate this defer there or declare the error up here.

Reply:
Given that there is no declaration of `err` before the `defer`, doesn't `err` in here refer to the returned error?

Reply:
Oh you are right! hadn't seen that 👍 

In this case I don't have anything to add then. Thank you